### PR TITLE
Pump rust-lightning to v0.0.108

### PR DESCRIPTION
**Pull request by meryacine**
This is needed because recent versions of rust-lightning [expose the `BigSize` struct](https://github.com/lightningdevkit/rust-lightning/commit/a58ae4c97b3436b743fd61a1fa697471131cf356#diff-1d6f48053419bbf9ae0bd220c527145558a5310885bb60c7bd5fef0b77814bee) (used in TLV encoding) which will be used in some messages in the tower's Lightning interface (check [bolt13](https://github.com/sr-gi/bolt13/blob/master/13-watchtowers.md)).

Related to #31

### CODE REVIEW

**Comment by sr-gi** [comment]
I guess we can bump this as part of your SoB main PR, or do you need it early for any other purpose?

**Comment by meryacine** [comment]
@sr-gi
Not for a specific thing, just separation of concerns.
These diffs look quite unrelated between the others.

**Comment by sr-gi** [comment]
Right. I'll merge it after the plugin release then.

File: teos/src/responder.rs [hunk 1]
```diff
@@ -493,17 +493,36 @@ impl chain::Listen for Responder {
     /// data deletion is performed accordingly. Moreover, lack of confirmations is check for the tracked transactions and
     /// rebroadcasting is performed for those that have missed too many.
     fn block_connected(&self, block: &bitcoin::Block, height: u32) {
-        log::info!("New block received: {}", block.header.block_hash());
+        let txdata: Vec<_> = block.txdata.iter().enumerate().collect();
+        self.filtered_block_connected(&block.header, &txdata, height);
+    }
+
+    /// Handles reorgs in the [Responder].
```
Comment (sr-gi) [inline teos/src/responder.rs:500]: Same as with the gatekeeper. This can be simply renamed and adapted.

File: teos/src/watcher.rs [hunk 2]
```diff
@@ -797,18 +797,37 @@ impl chain::Listen for Watcher {
     /// This also takes care of updating the [LocatorCache] and removing outdated data from the [Watcher] when
     /// told by the [Gatekeeper].
     fn block_connected(&self, block: &Block, height: u32) {
```
Comment (sr-gi) [inline teos/src/watcher.rs:799]: Same as with the two previous comments.

File: teos-common/src/appointment.rs [hunk 3]
```diff
@@ -1,8 +1,6 @@
 //! Logic related to appointments shared between users and the towers.
 
-use hex;
 use serde::{Deserialize, Serialize};
-
```
Comment (sr-gi) [inline teos-common/src/appointment.rs:5]: This is completely unrelated, isn't it?

**Review by sr-gi — CHANGES_REQUESTED** [review]
Check comments inline.

You can check some of the proposed changes in more detail here: https://github.com/sr-gi/rust-teos/commit/a6cc63a66833ca36af5a9f3b2ead09db2ac99051

File: teos-common/src/appointment.rs (hunk 3 above)
Comment (meryacine) [reply teos-common/src/appointment.rs:5]: Yes, nothing to do with the bump.

File: teos/src/dbm.rs [hunk 4]
```diff
@@ -533,7 +532,8 @@ impl DBM {
     /// When a new key is generated, old keys are not overwritten but are not retrievable from the API either.
     pub fn store_tower_key(&self, sk: &SecretKey) -> Result<(), Error> {
         let query = "INSERT INTO keys (key) VALUES (?)";
-        self.store_data(query, params![sk.to_string()])
+        let sk_bytes = Vec::from(sk.secret_bytes());
```
[the remark this answers is not in the document]
Comment (meryacine) [reply teos/src/dbm.rs]: It indeed breaks the older storage :(

[the remark this answers is not in the document]
Comment (sr-gi) [reply teos/src/dbm.rs]: No problem, turns out it can actually be adapted:

 ```
self.store_data(query, params![sk.display_secret().to_string()])
```

Then, `DBM::load_tower_key` will need to be rolled back to load a `String`.

**Review by sr-gi — CHANGES_REQUESTED** [review]
Update key load / store based on https://github.com/talaia-labs/rust-teos/pull/79#discussion_r933201224.

381540273bf82d86a1f5b932eef608060ffed518 message should be updated to properly reflect what the commit is doing (given the logic has changed).

Apart from that, LGTM.

File: watchtower-plugin/src/dbm.rs [hunk 5]
```diff
@@ -131,7 +131,6 @@ impl DBM {
                 "SELECT key FROM keys WHERE id = (SELECT seq FROM sqlite_sequence WHERE name=(?))",
             )
             .unwrap();
-
```
Comment (sr-gi) [inline watchtower-plugin/src/dbm.rs:134]: nit: add this back for smaller diffs

File: teos-common/Cargo.toml [hunk 6]
```diff
@@ -20,8 +20,8 @@ rand = "0.8.4"
 chacha20poly1305 = "0.8.0"
 
 # Bitcoin and Lightning
-bitcoin = { version = "0.27", features = [ "use-serde" ] }
-lightning = "0.0.105"
+bitcoin = { version = "0.28.0", features = [ "use-serde" ] }
+lightning = "0.0.108"
 
 [build-dependencies]
-tonic-build = "0.6"
```
Comment (sr-gi) [inline teos-common/Cargo.toml:27]: nit: There's a newline being added here, not sur why gh does not show it. Can you remove it?

**Review by sr-gi — CHANGES_REQUESTED** [review]
Just a couple of last nits.

On top of that, `test_block_connected` should be renamed to `test_filtered_block_connected` now for the components that implement it.

**Review by sr-gi — APPROVED** [review]
LGTM

Needs rebasing.
